diff -ruN geopandas-1.1.1/pyproject.toml geopandas-1.1.1-patch/pyproject.toml
--- geopandas-1.1.1/pyproject.toml	2025-06-26 23:04:34.000000000 +0200
+++ geopandas-1.1.1-patch/pyproject.toml	2025-10-06 13:21:03.866733363 +0200
@@ -22,10 +22,8 @@
 requires-python = ">=3.10"
 dependencies = [
     "numpy >= 1.24",
-    "pyogrio >= 0.7.2",
     "packaging",
     "pandas >= 2.0.0",
-    "pyproj >= 3.5.0",
     "shapely >= 2.0.0",
 ]
 
